ripe/atlas/cousteau/source.py

```diff
@@ -73,6 +77,37 @@ def set_type(self, value):
     doc_type = "Defines the type of probe's source."
     type = property(get_type, set_type, doc=doc_type)
 
+    # tags attribute
+    def get_tags(self):
+        """Getter for tags attribute"""
+        return self._tags
+
+    def set_tags(self, value):
+        """Setter for tags attribute"""
+        log = "Sources fields 'tags' should be a dict in the format " \
+              "{ \"include\": [ \"tag1\", \"tag2\", \"tagN\" ], " \
+              "\"exclude\": [ \"tag1\", \"tag2\", \"tagN\" ] }"
+
+        if not isinstance(value, dict):
+            raise MalFormattedSource(log)
+
+        for e in value:
+            if e not in ["include", "exclude"]:
+                raise MalFormattedSource(log)
+
+        for inc_exl in ["include", "exclude"]:
+            if inc_exl not in value:
+                raise MalFormattedSource(log)
+            if not isinstance(value[inc_exl], list):
+                raise MalFormattedSource(log)
+            if [e for e in value[inc_exl] if not isinstance(e, str)]:
+                raise MalFormattedSource(log)
+
```

Comment:
The first if has already been checked with the above condition right?
The other two you can play with the values of dict. something like:
```python
for tag_list in value.values():
    if not isinstance(tag_list, list):
       raise MalFormattedSource(log)
    if [tag for tag in tag_list if not isinstance(tag, str)]:
        raise MalFormattedSource(log)
```

When in python I generally try to avoid one letter variables and prefer something that make the one that read the code understand what it refers to. Code is read more times than written so it's important to make others understand easier :).